### PR TITLE
Add dask.bag.from_url

### LINKED ISSUE #242: Bag.from_url
Similar to #241 we might want to construct bags from resources found on the web.  The following lines from githubarchive.org show a possible api using unix-style options with ranges.

* Activity for 1/1/2015 @ 3PM UTC	`wget http://data.githubarchive.org/2015-01-01-15.json.gz`
* Activity for 1/1/2015	`wget http://data.githubarchive.org/2015-01-01-{0..23}.json.gz`
* Activity for all of January 2015	`wget http://data.githubarchive.org/2015-01-{01..30}-{0..23}.json.gz`

Is there a simple python way to duplicate this behavior?  Is `urllib2.urlopen` the best way to pull down data?

### PR DESCRIPTION
**Pull request by cowlicks**
partially closes #242 

This does not yet have the unix style ranges mentioned in the original issue. I have not tried adding them yet. I just wanted to get this initial prototype out because it seems useful on its own.

### CODE REVIEW

File: dask/bag/tests/test_bag.py [hunk 1]
```diff
@@ -190,6 +190,16 @@ def test_can_use_dict_to_make_concrete():
     assert isinstance(dict(b.frequencies()), dict)
 
 
+def test_from_url():
+    a = db.from_url(['http://google.com', 'http://github.com'])
+    assert a.npartitions == 2
+    a.compute()
```
Comment (mrocklin) [inline dask/bag/tests/test_bag.py]: Should probably test something here.  Maybe connect up to the dask README on github and assert that `dask` is in the top five lines using take?

File: dask/bag/core.py [hunk 2]
```diff
@@ -846,6 +851,27 @@ def from_sequence(seq, partition_size=None, npartitions=None):
...
+    2
+
+    """
+    if isinstance(urls, str):
+        urls = [urls]
+    dsk ={}
+    for i, u in enumerate(urls):
+        get_url = (urlopen, u)
+        dsk[('load-url', i)] = (lambda x: x.read(), get_url)
+    return Bag(dsk, 'load-url', len(urls))
```
Comment (mrocklin) [inline dask/bag/core.py]: Names need to be unique.  Sometimes we merge multiple bags and overlapping keys will clobber each other.

File: dask/bag/core.py [hunk 3]
```diff
@@ -846,6 +851,27 @@ def from_sequence(seq, partition_size=None, npartitions=None):
...
+    >>> b.npartions
+    2
+
+    """
+    if isinstance(urls, str):
+        urls = [urls]
+    dsk ={}
+    for i, u in enumerate(urls):
+        get_url = (urlopen, u)
+        dsk[('load-url', i)] = (lambda x: x.read(), get_url)
```
Comment (mrocklin) [inline dask/bag/core.py]: Suggest 

    dsk[('load-url', i)] = (list, (urlopen, u))

urlopen probably gives back an iterator of lines.  We should take advantage of this laziness.  The list will be taken out if we can manage it.  This keeps things lazy and probably gives us better reaction speeds on large files.

File: dask/bag/core.py [hunk 4]
```diff
@@ -846,6 +851,27 @@ def from_sequence(seq, partition_size=None, npartitions=None):
     return Bag(d, name, len(d))
 
 
+def from_url(urls):
+    """Create a dask.bag from a url
+
+    >>> a = from_url('http://google.com')
+    >>> a.npartitions
+    1
```
Comment (mrocklin) [inline dask/bag/core.py]: Suggest pulling a small bit of text from a project gutenberg book or some website that is both identifiable and that we expect to be around for a while.  I've used "It was the best of times, It was the worst of times" from dickens a fair amount.  It's easily recognizable.

File: dask/bag/tests/test_bag.py (hunk 1 above)
Comment (cowlicks) [inline dask/bag/tests/test_bag.py]: `a.take(n)` currently returns the first `n` characters. It should return lines.

Comment (mrocklin) [inline dask/bag/tests/test_bag.py]: I think that if we return the result from urlopen rather than urlopen(...).read() then we'll get this behavior.

Comment (mrocklin) [inline dask/bag/tests/test_bag.py]: I've found that having lines as items in the bag rather than characters also tends to be more useful in applications.

Comment (cowlicks) [inline dask/bag/tests/test_bag.py]: Using `urlopen` the way you suggested fixes this.

**Comment by mrocklin** [comment]
Lookin' good.  I'd still suggest having a sexier docstring with an actual example but otherwise I'd say this is good to go.

**Comment by cowlicks** [comment]
Gutenberg starts giving captchas after a few requests :(. I'll just use dask's README

File: dask/bag/core.py [hunk 5]
```diff
@@ -846,6 +851,29 @@ def from_sequence(seq, partition_size=None, npartitions=None):
     return Bag(d, name, len(d))
 
 
+def from_url(urls):
+    """Create a dask.bag from a url
+
+    >>> a = from_url('http://raw.githubusercontent.com/ContinuumIO/dask/master/README.rst')
+    >>> a.npartitions
+    1
+    >>> 'Dask\n' in a.take(1)
```
Comment (mrocklin) [inline dask/bag/core.py]: This looks more like a test than an example.  How about taking more from the README?

```python
>>> a.take(8)
Dask
====

|Build Status| |Coverage| |Doc Status| |Gitter|

Dask provides multi-core execution on larger-than-memory datasets using blocked
algorithms and task scheduling.  It maps high-level NumPy and list operations
```

**Comment by mrocklin** [comment]
This looks good.  Merging.

**Pull request merged**